tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -1985,6 +1960,185 @@ def visit_importfrom(self, node):
                         confidence=None,
                     )
 
+
+class TypePropertyNameTooLong(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that the character length of type and property names are not over X characters."""
+    name = "name-too-long"
+    priority = -1
+    msgs = {
+        "C4751": (
+            "Name is over standard character length of 40.",
+            "name-too-long",
+            "Only use names that are less than 40 characters."
```

Comment:
Is this actually in the guidelines?

Reply:
Related issue: https://github.com/Azure/azure-sdk-for-python/issues/26640

Was 40 chars the number that proved to have best signal:noise?

Reply:
40 was enough that it wasn't flagging everything, but it still catches unnecessarily long names